msys/members/views.py

```diff
@@ -473,7 +479,7 @@ def checkCard(request, card_rfid):
     try:
         card = AccessCard.objects.get(unique_id=card_rfid)
 
-        return cardDetails(request, card.pk)
+        return request, card
```

Comment:
Now that this function gets wrapped, what about when there is a AccessCard.DoesNotExist exception? Is there an potential problem with returning render(request, ...)?

Reply:
```
try:
        request, card = confirmCard(request, card_rfid)
    except ValueError:
        request = confirmCard(request, card_rfid)
        return request
```

Reply:
That's how I try to get around it, it might not be the fanciest (nor most versatile) solution, but it worked so far!. However, if you do think we should not implement it, just let me know. and off it'll go!

Reply:
Sure, let me see what can I do then, and I'll let you know right away

Reply:
```
request, card = confirmCard(request, card_rfid)
    if(card == 0):
        return request
```
I also initialized the card variable on the exception generated by confirmCard if it doesn't find the card. If there's no card, it'll be equal to 0

Reply:
Is that okay? It's a quick fix but It also works fine